### PR TITLE
Page Validator/Checker (#7)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**

### CODE REVIEW

File: taipy/gui/server.py [hunk 1]
```diff
@@ -143,7 +147,14 @@ def _render_page(self) -> t.Any:
         if page is None:
             return (jsonify({"error": "Page doesn't exist!"}), 400, {"Content-Type": "application/json; charset=utf-8"})
         # TODO: assign global scopes to current scope if the page has been rendered
-        page.render()
+        with warnings.catch_warnings(record=True) as w:
+            page.render()
+            if len(w) > 0:
+                print(message:= f"--- {len(w)} warnings were found for page {page.route} ---")
```
Comment (FabienLelaquais) [inline taipy/gui/server.py:151]: You will want to use the Flush parameter set to True in order to timely generate the messages.
In the console, 'regular' errors appear with a specific color (red for errors). Do you think we can mimic that (using the old \033[ trick)?

File: taipy/gui/page.py [hunk 2]
```diff
@@ -29,9 +30,13 @@ def __init__(self):
         self.route: t.Union[str, None] = None
         self.head: t.Union[str, None] = None
 
-    def render(self):
+    def render(self) -> None:
         if self.renderer is None:
-            raise RuntimeError(f"Can't render JSX for {self.route} due to missing renderer!")
+            warnings.warn(f"Can't render JSX for {self.route} due to missing renderer!")
```
Comment (FabienLelaquais) [inline taipy/gui/page.py:36]: No need to expose JSX in this message. Please make it simpler for users that don't know or care. Usually both.

File: taipy/gui/renderers/__init__.py [hunk 3]
```diff
@@ -39,15 +38,35 @@ def __process_content(self, content: str) -> None:
...
 
+class PageValidator(ABC):
+    """
+    This base class provide validation functionality to PageRenderer. This allow users to catch potential errors in runtime.
+    """
+
+    def __init__(self, page: PageRenderer) -> None:
+        self._page = page
+    
+    @abstractmethod
```
Comment (FabienLelaquais) [inline taipy/gui/renderers/__init__.py:80]: Hum. True/False loos rough to me, although I'm not sure we need anything else (like the rendered page).
What I do know is that we will need to review our errors/warnings generation in the parsers in order to let things through (like unclosed block tags) and others not that much (like unbound variables).

**Review by FabienLelaquais — APPROVED** [review]
Good first steps!

File: taipy/gui/page.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/page.py:36]: Thanks

Comment (FabienLelaquais) [reply taipy/gui/page.py:36]: Sorry my message was not clear.
I would prefer f"Cannot render page {self.route}: no renderer found"

Comment (dinhlongviolin1) [reply taipy/gui/page.py:36]: EVen less work for me now

**Comment by dinhlongviolin1** [comment]
Replaced by another pr

**Pull request closed without merging**